src/clr-angular/utils/focus/key-focus/key-focus.spec.ts

```diff
@@ -280,7 +292,19 @@ describe('KeyFocus directive', () => {
       // An example of this is the Tabs component.
       domComponent.buttons = Array.from(fixture.nativeElement.querySelectorAll('button'));
       fixture.detectChanges();
-      expect(clarityDirective.focusableItems.length).toBe(2);
+      expect(clarityDirective.focusableItems.length).toBe(3);
+    });
+    it('current value updates, when elements are being removed', () => {
```

Comment:
This is just a duplicate test of the above using the template instead of the API. It can be removed.

Reply:
It is not an exact duplicate. 
In this scenario, the changes are handled by the `focusableItems` setter.
In the case of the API test, the changes are handled by the `listenForItemUpdates()` method.
In my opinion, both tests are needed.

Reply:
What is triggering both tests to actually change is simply the toggling of `showLast`, which triggers the `listenForItemUpdates` to fire based on the QueryList change. These tests are just setting up the menu in two different ways ,but both asserting the result the same way (hence the duplication). 

Thinking about it more, this test should not be asserting that `current` is set but that the `document.activeElement` equals the expected button. If you change the test to assert that, I would be satisfied that its testing two different ways.

Reply:
Thanks. The test has been updated with the suggested assertions.